node/bft/src/primary.rs

```diff
@@ -1304,12 +1308,25 @@ impl<N: Network> Primary<N> {
         // whereby Narwhal requires 2f+1, however the BFT only requires f+1. Without this check, the primary
         // will advance to the next round assuming f+1, not 2f+1, which can lead to a network stall.
         let is_behind_schedule = is_quorum_threshold_reached && batch_round > self.current_round();
+
         // Check if our primary is far behind the peer.
         let is_peer_far_in_future = batch_round > self.current_round() + self.storage.max_gc_rounds();
-        // If our primary is far behind the peer, update our committee to the batch round.
-        if is_behind_schedule || is_peer_far_in_future {
+
+        // If our primary is behind shedule, update our committee to the batch round.
+        if is_behind_schedule {
             // If the batch round is greater than the current committee round, update the committee.
             self.try_increment_to_the_next_round(batch_round).await?;
+        // If our peer is far ahead, check if a quorum of peers is ahead and consider updating our committee.
+        } else if is_peer_far_in_future {
+            // Get the highest round seen from a quorum of the current committee
+            let committee = self.ledger.get_committee_for_round(self.current_round())?;
```

Comment:
This should be the `get_previous_committee_for_round`.

In the `mainnet` branch this should be changed to `get_committee_lookback_for_round`.

Reply:
e53bdb0e4

Let me know if you need me to create a new PR targeting mainnet.